cpp/ql/src/semmle/code/cpp/ir/dataflow/internal/DataFlowUtil.qll

```diff
@@ -389,6 +389,37 @@ private class ExplicitSingleFieldStoreQualifierNode extends PartialDefinitionNod
   }
 }
 
+private class ArrayStoreNode extends PartialDefinitionNode {
+  override ChiInstruction instr;
+  PointerAddInstruction add;
```

Comment:
Do I understand it correctly that this class and `PointerStoreNode` have to correspond to the implementation of `arrayStoreStepChi`? If so, we have to make that clear for future maintainers. Either they can share code or there can at least be a comment in all three places referring to each other.

Reply:
Yes, this is correct. I've added a couple of comments in 50ad4cfec4e80387fa67e8f9a21eeb45128a20fd to make this correspondence explicit.